docker/test/integration/minifi/core/SingleNodeDockerCluster.py

```diff
@@ -32,57 +38,96 @@ def __init__(self):
         # Get docker client
         self.client = docker.from_env()
 
-    def deploy_flow(self,
-                    flow,
-                    name=None,
-                    vols=None,
-                    engine='minifi-cpp'):
+    def __del__(self):
+        """
+        Clean up ephemeral cluster resources
+        """
+
+        # Clean up containers
+        for container in self.containers.values():
+            logging.info('Cleaning up container: %s', container.name)
```

Comment:
Removed. Also I see that it is ugly as hell to do part of the deletion outside the object that is expected to manage docker container lifetimes. However, if I add deletion here, I would still be able to access the container via the python API, but if if I tried calling delete on it, it would crash. I wonder if there is a way to sync up the list of available containers.